layer/components/Amount/Display.vue

```diff
@@ -0,0 +1,109 @@
+<script setup lang="ts">
+import { BigNumberInBase } from "@injectivelabs/utils";
+
+const props = withDefaults(
+  defineProps<{
+    amount: string | number | BigNumberInBase;
+    decimals?: number;
+    useSubscript?: boolean;
+    useAbbreviation?: boolean;
+    abbreviationThreshold?: number;
+    noTrailingZeros?: boolean;
+    subscriptThresholdDecimals?: number;
+    subscriptDecimals?: number;
+  }>(),
+  {
+    decimals: 6,
+    useSubscript: false,
+    useAbbreviation: false,
+    abbreviationThreshold: 1_000_000,
+    noTrailingZeros: false,
+    subscriptThresholdDecimals: 4,
+    subscriptDecimals: 4,
+  },
+);
+
+function abbreviateNumber(num: number, decimals: number = 2): string {
+  const suffixes = ["", "K", "M", "B", "T", "Q", "R", "S", "D", "N"];
+  const tier = (Math.log10(Math.abs(num)) / 3) | 0;
+  const suffix = suffixes[tier];
+  const scale = Math.pow(10, tier * 3);
+
+  const scaled = num / scale;
+  return scaled.toFixed(decimals) + (suffix || "");
+}
+
+const abbreviatedAmount = computed(() => {
+  const amount = new BigNumberInBase(props.amount || 0);
+
+  const nIsBiggerThanThreshold = amount.gte(props.abbreviationThreshold);
+
+  const minDecimalThreshold = new BigNumberInBase(1).div(
+    Math.pow(10, props.decimals),
+  );
+
+  const nIsLowerThanDecimalThreshold = amount.lt(minDecimalThreshold);
+
+  if (props.useAbbreviation && nIsBiggerThanThreshold) {
+    return "~" + abbreviateNumber(Number(props.amount || 0));
```

Comment:
can consider to use this segment of `useSharedBigNumberFormatter` for conditional `≈`
```
  const abbreviatedValueMatchesInput = new BigNumberInBase(number).eq(
    unAbbreviateNumber(abbreviatedValue) || '0'
  )

  return abbreviatedValueMatchesInput
    ? abbreviatedValue
    : `≈${abbreviatedValue}`
```